server/src/main/java/org/opensearch/identity/IdentityService.java

```diff
@@ -45,10 +48,10 @@ public IdentityService(final Settings settings, final List<IdentityPlugin> ident
     }
 
     /**
-     * Gets the current Subject
+     * Gets the current UserSubject
      */
-    public Subject getSubject() {
-        return identityPlugin.getSubject();
+    public UserSubject getUserSubject() {
```

Comment:
I think this is not correct: there could be at most one `Subject` active in any more of time (user or plugin or extension or ...), the `Subject getSubject()` is the way to access it. Please revert.

Reply:
Changed back to `getCurrentSubject`. See comment here: https://github.com/opensearch-project/OpenSearch/pull/14630#discussion_r1722088100